.rubocop.yml

```diff
@@ -0,0 +1,204 @@
+---
+require:
+  - rubocop-performance
+  - rubocop-rails
+  - rubocop-rspec
+
+# Built-in config:
+# https://github.com/bbatsov/rubocop/blob/master/config/default.yml
+
+Rails:
+  Enabled: true
+
+AllCops:
+  NewCops: enable
+  DisplayCopNames: true
+  DisplayStyleGuide: true
+  Exclude:
+    - 'tmp/**/*'
+    # - Gemfile
+    - 'node_modules/**/*'
+
+Style/Documentation:
+  Enabled: false
+
+Layout/LineLength:
+  Max: 200
+
+Metrics/BlockLength:
+  Max: 50
+
+Metrics/CyclomaticComplexity:
+  Max: 10
+
+Metrics/MethodLength:
+  Max: 50
+
+Style/GlobalVars:
+  Enabled: false
```

Comment:
Highlighting this change because it might warrant some discussion. We are using a global var for the config instance to save passing it around. I guess we could refactor it to something nice but I went with the smallest change possible.

Reply:
i'd maybe prefer disabling in place, just to discourage any other global vars showing up?

Reply:
It ended up a bit ugly because the Rubocop disabling comments were required in some of the generated code. Instead I have replaced it with a constant called `TEMPLATE_CONFIG`. Using a constant like this doesn't feel super idiomatic but it seems equivalent to what we had before and it keeps Rubocop happy :shrug:

Reply:
fwiw you can also configure Rubocop to allow specific globals: https://docs.rubocop.org/rubocop/cops_style.html#configurable-attributes-35